provisioning/tools/create_test_cert.js

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var fs = require('fs');
+var pem = require('pem');
+var chalk = require('chalk');
+var camelcase = require('camelcase');
+var argv = require('yargs')
+          .command('root [commonName]', 'create a root certificate', (yargs) => {
+            yargs.positional('commonName', { describe: 'name to use for cert', default: 'Test Root Cert' })
+          })
+          .command('intermediate <commonName> <parentCommonName>', 'create an intermediate cert', (yargs) => {
+            yargs.positional('commonName', { describe: 'name to use for cert' })
+            yargs.positional('parentCommonName', { describe: 'common name of parent cert' })
+          })
+          .command('device <commonName> [parentCommonName]', 'create a device cert', (yargs) => {
+            yargs.positional('commonName', { describe: 'name to use for cert' })
+            yargs.positional('parentCommonName', { describe: 'common name of parent cert' })
+          })
+          .command('verification <certToVerify> <verificationCode>', 'provide proof of possession for a cert using a verification code', (yargs) => {
+            yargs.positional('certToVerify', { describe: 'common name for certificate to verify' })
+            yargs.positional('verificationCode', { describe: 'verification code from portal' })
+          })
+          .demandCommand()
+          .help()
+          .argv;
```

Comment:
asking for later: how much of this would belong in the azure iot cli or in iothub-explorer? could be useful to other SDKs, right? also maybe for demo purposes for the field?

Reply:
also maybe in the future an option for self-signed?

---
In reply to: [161131683](https://github.com/Azure/azure-iot-sdk-node/pull/198#discussion_r161131683) [](ancestors = 161131683)

Reply:
self signed is included under the device command.  This is documented in the readme, but a "selfsigned" command might be better.

---
In reply to: [161131796](https://github.com/Azure/azure-iot-sdk-node/pull/198#discussion_r161131796) [](ancestors = 161131796,161131683)